src/storage/src/source/generator.rs

```diff
@@ -136,28 +138,25 @@ impl GeneratorKind {
         committed_uppers: impl futures::Stream<Item = Antichain<MzOffset>> + 'static,
         start_signal: impl std::future::Future<Output = ()> + 'static,
     ) -> (
-        StackedCollection<G, (usize, Result<SourceMessage, DataflowError>)>,
+        BTreeMap<GlobalId, StackedCollection<G, Result<SourceMessage, DataflowError>>>,
         Option<Stream<G, Infallible>>,
         Stream<G, HealthStatusMessage>,
         Stream<G, ProgressStatisticsUpdate>,
         Vec<PressOnDropButton>,
     ) {
         // figure out which output types from the generator belong to which output indexes
         let mut output_map = BTreeMap::new();
-        for (_, export) in config.source_exports.iter() {
-            let output_type = match &export.export.details {
+        for (idx, (_, export)) in config.source_exports.iter().enumerate() {
+            let output_type = match &export.details {
                 SourceExportDetails::LoadGenerator(details) => details.output,
                 // This is an export that doesn't need any data output to it.
                 SourceExportDetails::None => continue,
-                _ => panic!(
-                    "unexpected source export details: {:?}",
-                    export.export.details
-                ),
+                _ => panic!("unexpected source export details: {:?}", export.details),
             };
             output_map
                 .entry(output_type)
                 .or_insert_with(Vec::new)
-                .push(export.ingestion_output);
+                .push(idx);
```

Comment:
The idea in all of these is that indexing things by a `usize` is an implementation detail that sources can choose to do. So anything that was previously using the `ingestion_output` thing has been switched to be using the index of the source as found in the `source_exports` map